### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by DigitalSmile**
Just changed minor things to have camel case, package order and some todo items.
Moved examples into unit tests

### CODE REVIEW

File: src/main/java/com/bazaarbot/agent/BasicAgent.java [hunk 1]
```diff
@@ -3,183 +3,167 @@
...
 
-    protected double determinePurchaseQuantity(Market bazaar, ICommodity commodity_) {
-        Double mean = bazaar.getAverageHistoricalPrice(commodity_,_lookback);
+
+    protected double determinePurchaseQuantity(int observeWindow, double averageHistoricalPrice, ICommodity commodity) {
+        if (averageHistoricalPrice <= 0) {
+            return 0;
+        }
+        CommodityPricingRange tradingRange = observeTradingRange(commodity, observeWindow);
+        double favorability = tradingRange == null ? 1 : tradingRange.positionInRange(averageHistoricalPrice);
```
Comment (DigitalSmile) [inline src/main/java/com/bazaarbot/agent/BasicAgent.java:107]: I doubt about this line... Why we need something in historical pricing range if we do not need this?..

File: src/main/java/com/bazaarbot/agent/BasicAgent.java [hunk 2]
```diff
@@ -3,183 +3,167 @@
...
-            //do 1 - favorability to see how close we are to the low end
-            double amount_to_buy = Math.round(favorability * _inventory.shortage(commodity_));
-            //double
-            if (amount_to_buy < 1)
-            {
-                amount_to_buy = 1;
-            }
-             
-            return amount_to_buy;
+        if (amountToBuy < 1) {
```
Comment (DigitalSmile) [inline src/main/java/com/bazaarbot/agent/BasicAgent.java:113]: And this one is driving me crazy O_o

**Comment by JeremyWildsmith** [comment]
For agent ID, it's been changed to use a hash of a uuid; there could be a hash collision here (though unlikely)

Instead we could use a private static final AtomicInt and get/increment it to assign the agent I'd?

Edit: also do we need an agentId? I wonder what it's actually being used for?
Edit: Looks like it literally just used to indirectly reference agent. Lets change it to a reference to agent. I can pull and make those changes.

**Comment by DigitalSmile** [comment]
We need to think first about why do we really need the id... There are pros and cons on storing object reference...
As for the collision, it would be really unlikely, but this is the fastest way to get pseudo random and unique integer.

**Comment by JeremyWildsmith** [comment]
Looks like this merge caused the unit tests too fail; so I had to revert it. Sorry, I should've checked before merging it in.

**Comment by JeremyWildsmith** [comment]
Okay I have setup travis CI so we can catch issues like this earlier. Sorry for the inconvenience.

BTW I did a git bisect on your repo, it first started failing on commit #e66a7dbbc1224cdf3ccbc6917480a8dce8f6c5e2 due to commenting out the unused variable (also commented out a function call)
However I don't know how to fix it later in the commits.

**Pull request merged**